tls/irc_client_tls.ml

```diff
@@ -0,0 +1,48 @@
+open Lwt.Infix
+
+module Io = struct
+  type 'a t = 'a Lwt.t
+  let (>>=) = Lwt.bind
+  let return = Lwt.return
+
+  type file_descr = {
+    ic: Tls_lwt.ic;
+    oc: Tls_lwt.oc;
+  }
+
+  type inet_addr = string
+
+  type config = Tls.Config.client
+
+  let default_config : Tls.Config.client =
+    Tls.Config.client ~authenticator:X509.Authenticator.null ()
```

Comment:
sorry for the lack of documentation.  I don't know whether this irc client has an interactive configuration or how it is configured.  In [jackline](https://github.com/hannesm/jackline), I have such an interactive way, and ask the user for either a certificate fingerprint or a (set of) trust anchors -- https://github.com/hannesm/jackline/blob/286c91e7f07c0ad17ca830b748a3c97b02541392/cli/cli_config.ml#L189-L234